src/WebJobs.Script/StorageProvider/IAzureTableStorageProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Data.Tables;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public interface IAzureTableStorageProvider
+    {
+        IConfiguration Configuration { get; }
+
+        bool TryCreateHostingTableServiceClient(out TableServiceClient client);
+
+        bool TryCreateTableServiceClientFromConnection(string connection, out TableServiceClient client);
```

Comment:
Should we use the same name for both methods? T`ryCreateTableServiceClient`. The second one is just another overload which takes the connection string.

Reply:
The reasoning behind that name was more semantic than technical. To be explicit about it would create a client used for our "Hosting" purposes a.k.a AzureWebJobsStorage client. I can clarify that in the summary of the method. What do you think?

Reply:
I am not sure the method name should indicate how the caller is going to use the TableServiceClient we return. So, I like using the `TryCreateTableServiceClient` for both.  This is not a blocker BTW.